packages/x-codemod/src/v8.0.0/pickers/preset-safe/index.ts

```diff
@@ -1,8 +1,10 @@
+import transformAdapterDateFnsImports from '../rename-adapter-date-fns-imports';
 import transformFieldValue from '../rename-and-move-field-value-type';
 
 import { JsCodeShiftAPI, JsCodeShiftFileInfo } from '../../../types';
 
 export default function transformer(file: JsCodeShiftFileInfo, api: JsCodeShiftAPI, options: any) {
+  file.source = transformAdapterDateFnsImports(file, api, options);
```

Comment:
WDYT, should we include it in the safe preset if it is NOT idempotent?
Running it repeatedly will change the imports back and forth. 🙈

Reply:
> WDYT, should we include it in the safe preset if it is NOT idempotent?

Oh crap, that's a good question.
Could it be work adding a small comment at the end of the line that we can check to make it idempotent? :grimacing:

Reply:
I don't imagine how we could do it cleanly (without adding some marker comment)... 🤷 
It's the nature of the rename, that we are essentially swapping the adapter names, with the exception that `AdapterV3` becomes `Adapter` and then after a subsequent run would become `AdapterV2` at which point the codemod would no longer rename it back to `Adapter`, but it's still:
```sh
`AdapterV3` -> run codemod -> `Adapter` -> run codemod -> `AdapterV2`
```

Reply:
> I don't imagine how we could do it cleanly (without adding some marker comment)... 🤷

Me neither
The only solution I see is to transform:

```ts
import { AdapterDateFnsV3 } from '@mui/x-date-pickers/AdapterDateFnsV3';
```

into

```ts
// x-codemod/v8.0.0/pickers/adapter-date-fns-imports makes sure this codemod is idempotent
import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFns'; 
```

Which is far from great :grimacing: